cwms-data-api/src/test/java/cwms/cda/data/dao/watersupply/WaterContractDaoTestIT.java

```diff
@@ -0,0 +1,386 @@
+        });
+    }
+
+    @Test
+    void testRenameWaterUser() throws Exception {
+        WaterUser newUser = buildTestWaterUser("TEST RENAME USER");
+        WaterUser updatedUser = buildTestWaterUser("TEST RENAME USER 2");
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(newUser, true);
+            dao.renameWaterUser(newUser.getEntityName(), updatedUser.getEntityName(), newUser.getProjectId());
+            WaterUser retrievedUser = dao.getWaterUser(newUser.getProjectId(), updatedUser.getEntityName());
+            DTOMatch.assertMatch(updatedUser, retrievedUser);
+        });
+        cleanupUserRoutine(updatedUser);
+    }
+
+    @Test
+    void testRenameWaterContract() throws Exception {
+        WaterUserContract oldContract = buildTestWaterContract("Test Old Name", true);
+        WaterUserContract renamedContract = buildTestWaterContract("Test New Name", true);
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(oldContract.getWaterUser(), true);
+            dao.storeWaterContract(oldContract, true, false);
+            WaterUser user = new WaterUser.Builder().withEntityName(oldContract.getWaterUser().getEntityName())
+                    .withProjectId(oldContract.getWaterUser().getProjectId())
+                    .withWaterRight(oldContract.getContractId().getName()).build();
+            dao.renameWaterContract(user, oldContract.getContractId().getName(),
+                    renamedContract.getContractId().getName());
+            List<WaterUserContract> retrievedContracts = dao.getAllWaterContracts(
+                    renamedContract.getWaterUser().getProjectId(), renamedContract.getWaterUser()
+                            .getEntityName());
+            assertFalse(retrievedContracts.isEmpty());
+            DTOMatch.assertMatch(retrievedContracts.get(0), renamedContract);
+        });
+        cleanupContractRoutine(renamedContract);
+        cleanupUserRoutine(renamedContract.getWaterUser());
+    }
+
+    @Test
+    void testDeleteWaterUser() throws Exception {
+        WaterUser newUser = buildTestWaterUser("TEST DELETE USER");
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            WaterContractDao dao = new WaterContractDao(ctx);
+            dao.storeWaterUser(newUser, true);
+            dao.deleteWaterUser(newUser.getProjectId(), newUser.getEntityName(), DELETE_ACTION);
+            assertNull(dao.getWaterUser(newUser.getProjectId(), newUser.getEntityName()));
```

Comment:
shouldn't this throw a not found exception?

Reply:
Added not found exception throws to getWaterUser and getWaterContract